File: clients/javascript/game/virus.js
const Vector = require('victor');

module.exports = class Virus {
  constructor(position, mass, radius) {
    this.position = position;
    this.mass = mass;
    this.radius = radius;
  }

  static parse(payload) {
    return new Virus(
      Vector.fromObject(payload.position),
      payload.mass,
      payload.radius
    );
  }
};

File: clients/java/src/main/java/io/aigar/game/models/Virus.java
package io.aigar.game.models;

import org.mini2Dx.gdx.math.Vector2;

public class Virus {
    private int radius;
    private int mass;
    private Vector2 position;

    public int getRadius() {
        return radius;
    }

    public void setRadius(int radius) {
        this.radius = radius;
    }

    public int getMass() {
        return mass;
    }

    public void setMass(int mass) {
        this.mass = mass;
    }

    public Vector2 getPosition() {
        return position;
    }

    public void setPosition(Vector2 position) {
        this.position = position;
    }
}
